src/app/containers/affiliation-form-dialog/index.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { SyntheticEvent, useCallback, useState } from 'react';
+import { TextField } from '@material-ui/core';
+import { useDispatch } from 'react-redux';
+import { set, cloneDeep } from 'lodash';
+
+import { useAffiliationFormStyles } from './styles';
+import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
+import * as manuscriptActions from 'app/actions/manuscript.actions';
+import { PromptDialog } from 'app/components/prompt-dialog';
+import { ActionButton } from 'app/components/action-button';
+import { Affiliation, createAffiliation } from 'app/models/affiliation';
+
+interface AffiliationFormDialogProps {
+  affiliation?: Affiliation;
+}
+
+const renderConfirmDialog = (title: string, msg: string, onAccept: () => void, onReject: () => void) => {
+  return (
+    <PromptDialog
+      title={title}
+      message={msg}
+      isOpen={true}
+      onAccept={onAccept}
+      onReject={onReject}
+      acceptLabel="Delete"
+      rejectLabel="Cancel"
+      acceptVariant="containedWarning"
+      rejectVariant="secondaryOutlined"
+    />
+  );
+};
+
+const labelProps = { shrink: true };
+
+export const AffiliationFormDialog: React.FC<AffiliationFormDialogProps> = (props) => {
+  const dispatch = useDispatch();
+  const isNewAffiliation = !props.affiliation;
+
+  const [affiliation, setAffiliation] = useState<Affiliation>(
+    props.affiliation ||
+      createAffiliation(undefined, {
+        label: '',
+        institution: {
+          department: '',
+          name: ''
+        },
+        address: {
+          city: ''
+        },
+        country: ''
+      })
+  );
+  const [isConfirmShown, setConfirmSnow] = useState<boolean>(false);
```

Comment:
Typo? Guessing it should be setConfirmShow

Reply:
oops :) fixed